client/src/lib.rs

```diff
@@ -56,24 +56,49 @@ where
     CmdT: TpmCommand,
     T: Tpm,
 {
-    Ok(run_command_with_handles(cmd, CmdT::Handles::default(), CmdSessions::default(), tpm)?.0)
+    Ok(run_command_with_handles(cmd, CmdT::Handles::default(), (), tpm)?.0)
+}
+
+fn marshal_auth_size(auth_offset: usize, buffer: &mut [u8]) -> TssResult<usize> {
+    let auth_size = (auth_offset - size_of::<u32>()) as u32;
```

Comment:
This is correct, but probably worth a comment. Something to the effect of "From TPM2 Specification: 18.5: authorizationSize/parameterSize: The authorization size in the command structure does not contain the size of the integer for storing the `authorizationSize` itself."